src/PolarSSL++/BlockingSslClientSocket.cpp

```diff
@@ -182,7 +183,11 @@ bool cBlockingSslClientSocket::SetTrustedRootCertsFromString(const AString & a_C
 
 bool cBlockingSslClientSocket::Send(const void * a_Data, size_t a_NumBytes)
 {
-	ASSERT(m_IsConnected);
+	if (!m_IsConnected)
+	{
+		m_LastErrorText = "Socket is closed";
+		return false;
+	}
```

Comment:
Does it matter if the socket suddenly closes anywhere in this loop, after the check?